src/communication/hpxreceiver.h

```diff
@@ -37,9 +42,17 @@ class HPXReceiver : public hpx::components::simple_component_base<HPXReceiver<CA
 
     static hpx::future<boost::shared_ptr<HPXReceiver> > make(const std::string& name, std::size_t rank = 0)
     {
```

Comment:
Needs to be discoverable from remote localities. It'll receive data from remote endpoints and local objects will then retrieve the data through the shared_ptr.